### PR TITLE
Add Dell Emc Networking Sonic rest support

### PR DESCRIPTION
**Pull request by javeedf**
Add Dell Emc Networking Sonic rest support

### CODE REVIEW

File: plugins/module_utils/network/common/network.py [hunk 1]
```diff
@@ -236,7 +236,7 @@ def get_resource_connection(module):
 
     capabilities = get_capabilities(module)
     network_api = capabilities.get("network_api")
-    if network_api in ("cliconf", "nxapi", "eapi", "exosapi"):
+    if network_api in ("cliconf", "nxapi", "eapi", "exosapi", "sonic_rest"):
```
Comment (ganeshrn) [inline plugins/module_utils/network/common/network.py:239]: I think we should default to using `Connection` class. Adding an entry here for each transport variant is not scalable.
```
    if network_api == "netconf":
        module._connection = NetconfConnection(module._socket_path)
    elif network_api == "local":
        # This isn't supported, but we shouldn't fail here.
        # Set the connection to a fake connection so it fails sensibly.
        module._connection = LocalResourceConnection(module)
    else:
        module._connection = Connection(module._socket_path)
```

Comment (javeedf) [reply plugins/module_utils/network/common/network.py]: updated as suggested

**Comment by javeedf** [comment]
@ganeshrn Could you please review and let us know if this is ok?

**Comment by ganeshrn** [comment]
CI failing for sanity test
```
ERROR: Found 2 pep8 issue(s) which need to be resolved:
ERROR: plugins/module_utils/network/common/network.py:239:1: W293: blank line contains whitespace
ERROR: plugins/module_utils/network/common/network.py:248:1: W293: blank line contains whitespace
```

and unit tests https://dashboard.zuul.ansible.com/t/ansible/build/1e11470187564d94b5d97479d0614d20/console

**Comment by ganeshrn** [comment]
Unit test failures are related to PR https://github.com/ansible/ansible-zuul-jobs/pull/549 and should be fixed after the PR is merged

**Comment by GomathiselviS** [comment]
recheck

**Comment by javeedf** [comment]
@GomathiselviS @pabelanger anything expected from me to merge this?

**Pull request merged**